gno.land/pkg/gnoweb/handler.go

```diff
@@ -27,37 +28,118 @@ type StaticMetadata struct {
 	Analytics  bool
 }
 
+type AliasKind int
+
+const (
+	GnowebPath AliasKind = iota
+	StaticMarkdown
+)
+
+type AliasTarget struct {
+	value string
+	kind  AliasKind
+}
+
 // WebHandlerConfig configures a WebHandler.
 type WebHandlerConfig struct {
-	Meta      StaticMetadata
-	WebClient WebClient
+	Meta             StaticMetadata
+	WebClient        WebClient
+	MarkdownRenderer *MarkdownRenderer
+	aliases          map[string]AliasTarget
+}
+
+// NewDefaultWebHandlerConfig initializes a [WebHandlerConfig] with default settings.
+func NewDefaultWebHandlerConfig() *WebHandlerConfig {
+	// Default aliases, can be overridden using '--aliases' flag.
+	aliases := map[string]AliasTarget{
+		"/":           {"/r/gnoland/home", GnowebPath},
+		"/about":      {"/r/gnoland/pages:p/about", GnowebPath},
+		"/gnolang":    {"/r/gnoland/pages:p/gnolang", GnowebPath},
+		"/ecosystem":  {"/r/gnoland/pages:p/ecosystem", GnowebPath},
+		"/start":      {"/r/gnoland/pages:p/start", GnowebPath},
+		"/license":    {"/r/gnoland/pages:p/license", GnowebPath},
+		"/contribute": {"/r/gnoland/pages:p/contribute", GnowebPath},
+		"/events":     {"/r/gnoland/events", GnowebPath},
+	}
+
+	return &WebHandlerConfig{
+		aliases: aliases,
+	}
+}
+
+// SetAliases parses the given aliases string and sets the aliases map.
+// Any value of form 'static:<file>' will be loaded as a static markdown file.
+func (cfg *WebHandlerConfig) SetAliases(aliasesStr string) error {
+	if strings.TrimSpace(aliasesStr) == "" {
+		return nil
+	}
+
+	// Split the aliases string by commas.
+	aliasEntries := strings.Split(aliasesStr, ",")
+
+	// Add each alias entry to the aliases map.
+	for _, entry := range aliasEntries {
+		parts := strings.Split(entry, "|")
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid alias entry: %s", entry)
+		}
+
+		// Trim whitespace from both parts.
+		parts[0] = strings.TrimSpace(parts[0])
+		parts[1] = strings.TrimSpace(parts[1])
+
+		// Check if the value is a path to a static file.
+		if strings.HasPrefix(parts[1], "static:") {
+			// If it is, load the static file content and set it as the alias.
+			staticFilePath := strings.TrimPrefix(parts[1], "static:")
+
+			content, err := os.ReadFile(staticFilePath)
+			if err != nil {
+				return fmt.Errorf("failed to read static file %s: %w", staticFilePath, err)
+			}
+			cfg.aliases[parts[0]] = AliasTarget{value: string(content), kind: StaticMarkdown}
+		} else { // Otherwise, treat it as a normal alias.
+			cfg.aliases[parts[0]] = AliasTarget{value: parts[1], kind: GnowebPath}
+		}
+	}
+
+	return nil
 }
```

Comment:
This is too opinionated and more adapted for the command line. If I want to initialize and use WebHandler from the outside, I want to be able to directly interact with WebHandlerConfig and update/modify AliasTarget map myself, not to construct a comma-separated list of arguments.
This should be on the same level as the cmd, ideally passing a list of aliases to the App and then to the Handler.